front-end/src/Pages/Profile.js

```diff
@@ -1,11 +1,16 @@
-import React from "react";
+import React, { useContext } from "react";
 import ProjectList from "../Components/ProjectList";
+import { UserContext } from "../UserContext";
 
 function Profile() {
+  const { user } = useContext(UserContext);
+
   //the projects will be passed props of what users profile we are on
+
   return (
```

Comment:
you need to define a state for null user.  should redirect to login

Reply:
profile will never render unless a user is defined. I have the navigation bar conditionally rendering login 
 OR logout/profile... also when login or logout is submitted I route the app back to the home page